src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -72,14 +72,30 @@ public override async Task<dynamic> GetCertificatesInResourceGroupAsync(string s
             return await GetRuntimeSiteSlotMapInternal(stampName, siteName, slotName);
         }
 
-        private async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMapInternal(string stampName, string siteName, string slotName)
+        public override async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(OperationContext<App> cxt, string stampName = null, string siteName = null, string slotName = null, DateTime? endTime = null)
+        {
+            if (cxt?.Resource?.Stamp == null)
+                throw new ArgumentNullException(nameof(cxt));
+
+            stampName = string.IsNullOrWhiteSpace(stampName) ? cxt.Resource.Stamp.InternalName : stampName;
+            siteName = string.IsNullOrWhiteSpace(siteName) ? cxt.Resource.Name : siteName;
+            endTime = endTime == null ? DateTime.SpecifyKind(DateTime.Parse(cxt.EndTime), DateTimeKind.Utc) : endTime;
+
+            return await GetRuntimeSiteSlotMapInternal(stampName, siteName, slotName, endTime);
+        }
+
+        private async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMapInternal(string stampName, string siteName, string slotName, DateTime? endTime = null)
         {
             if (string.IsNullOrWhiteSpace(stampName))
                 throw new ArgumentNullException(nameof(stampName));
             if (string.IsNullOrWhiteSpace(siteName))
                 throw new ArgumentNullException(nameof(siteName));
 
-            var result = await GetObserverResource($"stamps/{stampName}/sites/{siteName}/runtimesiteslotmap");
+            endTime = endTime == null ? DataProviderContext.QueryEndTime : endTime;
+            var endTimeUtcString = $"{endTime.Value.ToUniversalTime():O}";
```

Comment:
Whether the endTime comes from the OperationContext or the DataProviderContext, it will already be in UTC so perhaps you don't need to do the ToUniversalTime conversion.

Reply:
This has to be there for the overrides to work in case a detector passes in `DateTime.Now` or some other potentially-localized DateTime
```
$"{DateTime.Now.ToUniversalTime():O}"
"2020-02-04T21:46:33.9415616Z"
$"{DateTime.Now:O}"
"2020-02-04T13:46:39.4756676-08:00"
```